tests/integration/acp/query/avg_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package test_acp
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+	acpUtils "github.com/sourcenetwork/defradb/tests/integration/acp"
+)
+
+func TestACP_QueryAverageWithoutIdentity(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Test acp, query average without identity",
+
+		Actions: []any{
+			getSetupEmployeeCompanyActions(),
```

Comment:
> I cannot see what this test is really testing, unless I bounce between multiple files trying to hold different bits of info in my head and do the maths myself.

This confrontation will never end :)
I agree with your point that the tests needs to be broken up into smaller bits. But then the only thing that will make them different is a single request. And if I inline the setup part it will take 90% of the whole test body.

There is no need to "bounce between multiple files", it's one just to the setup method (potentially only once for all tests) and back.

Phrases like "It would be much easier to read" are highly subjective as it expresses your personal perception.
For me personally it would take much more time to read tests with "the stuff all in the same place", because all this 90% of copy&pasted code needs to be checked as well, right? Otherwise why is it there? It's good when you know the setup code is the same in all tests, but what if you don't?
With the current approach you know it immediately and you see right away what makes these tests differ from one another.

Reply:
> This confrontation will never end :)

I'll resist responding here then, as it will just bog the PR down :P  Maybe worth a discord thread after the release though.

Please document the tests as-is then, a few code comments explaining what is going on to anyone reading can still greatly improve things here.

Reply:
I added some comments when I though it would benefit a reader. Let me know if you think any other part needs more comments.